### DISCUSSION TITLE
$q.dialog({persistent: true}) does not work while using component

### DISCUSSION THREAD

**Discussion by sonicgamer0721** [quasarframework/quasar#14249] [General - Components / Directives / etc] [answered]
Hi guys, I made a component named AddRowForm, and used it in $q.dialog(), but "persistent: true" doesn't work, could you please help me?

```
const addRow = () => {
  $q.dialog({
    persistent: true,
    component: AddRowForm,

    // props forwarded to your custom component
    componentProps: {
      url: props.url,
      table,
      formComponent,
      // ...more..props...
    },
  }).onOk(async () => {
    await loadData();
  });
};
```

**Comment by pdanpdan** [comment]
You have to put persistent on the QDialog

**Reply by sonicgamer0721** [reply]
It seems that I can't use both component and persistent in $q.dialog() at the same time.
Do you mean I should give up $q.dialog() and use QDialog if I need persistent?

**Comment by pdanpdan** [comment, marked as answer]
When you use custom components for dialog you are rendering the QDialog in your component (AddRowForm).
So you have to use the `persistent` prop on it.
And in this case it makes no sense to pass the key in the call to `$q.dialog` because there is no place where it can be used automatically.

**Reply by sonicgamer0721** [reply]
OK. I got it. Thanks so much for your explanation. XD

**Comment by yusufkandemir** [comment]
https://quasar.dev/quasar-plugins/dialog#invoking-custom-component
> But in this case **you will be responsible for handling everything** (including your own component props).

```html
<template>
  <!-- Add `persistent` or any other QDialog prop here 👇 -->
  <q-dialog ref="dialogRef" persistent @hide="onDialogHide">
```

If you want to make it conditional, define it as a prop in your dialog component, then pass it through `componentProps` when calling `$q.dialog`.
```html
<template>
  <q-dialog ref="dialogRef" :persistent="persistent" @hide="onDialogHide">
```
```js
defineProps(['persistent'])
```
```
$q.dialog({
  component: ...,
  componentProps: {
    // ...
    persistent: true
  }
})
```
